src/widgets/ReportChart.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2018
+ */
+
+import React, { useState } from 'react';
+import { View, StyleSheet } from 'react-native';
+import PropTypes from 'prop-types';
+import { PieChart, BarChart, LineChart, ReportTable } from './index';
+
+export const ReportChart = ({ report }) => {
+  const { type, data } = report;
+  const { rows, header } = data;
+  const [dimensions, setDimensions] = useState({ height: null, width: null });
+  const { height, width } = dimensions;
+
+  // Victory Native sizes are set using absolute values. Parents dimensions are used to
+  // calculate relative values for width and height for each chart.
+  const onLayout = event => {
+    const newDimensionsObj = {
+      height: event.nativeEvent.layout.width,
+      width: event.nativeEvent.layout.height,
+    };
+    setDimensions(newDimensionsObj);
+  };
```

Comment:
I think if we're doing this here we should *only* do it here, not in the children. So, that would mean passing to the charts i.e.

```
const BarChart = ({width, height, report}) => {
 ....
}
```

and using that here :

```
case 'BarChart':
    return <BarChart report={report} width={width} height={height}/>
```

There doesn't seem to be a reason to do it twice

Alternatively, you can just keep it in the chart components, and have them do the check for width/height in all of them, but I think it's nicer to do it the one time in here

Reply:
I agree! that's what we discussed yesterday afternoon and totally happy with keeping that in the parent and take it out from children components.

Reply:
I will create a issue to remove width and height calculation on children components as well.

Reply:
ok sure!